aws/amazon-chime-sdk-js
# Log error if pass undefined to choose input API

**Issue #:**
#674: Prevent error exception when MediaDeviceInfo is not defined.
#851: Log error and return if device parameter is undefined.
#886: Update single.js package.json to use latest v2 version of amazon-chime-sdk-js.

**Testing**

1. Have you successfully run `npm run build:release` locally? Yes
2. How did you test these changes? 
- Unit tests
- Open a web browser and manually call chooseAudioInputDevice(undefined) and verify that an error is logged.
- Open a web browser and manually call delete MediaDeviceInfo and listAudioDevice and verify that there is no error thrown.
3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it?
No
4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? No
5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? No


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
